### PR TITLE
Feat/54/crud unit with full info

### PR DESCRIPTION
**Pull request by mershalova**

### CODE REVIEW

File: MilitaryManager.Core/Services/UnitService.cs [hunk 1]
```diff
@@ -58,7 +73,11 @@ public async Task<UnitDTO> UpdateUnitAsync(UnitDTO query)
 
         public async Task<UnitDTO> DeleteUnitAsync(int id)
         {
-            var unit = new Unit() { Id = id };
+            var unit = await _unitRepository.GetFirstBySpecAsync(new Units.UnitById(id));
```
Comment (vladtymo) [inline MilitaryManager.Core/Services/UnitService.cs]: Is it necessary to include all the related properties to the unit for deleting it?

**Review by vladtymo — CHANGES_REQUESTED** [review]

File: MilitaryManager.Core/Services/UnitService.cs (hunk 1 above)
Comment (mershalova) [reply MilitaryManager.Core/Services/UnitService.cs]: It only deletes the unit itself, no related properties are deleted

File: MilitaryManager.Units.API/Controllers/UnitController.cs [hunk 2]
```diff
@@ -32,6 +32,20 @@ public async Task<IActionResult> GetTreeNode([FromRoute] int id)
...
 
+        [HttpGet]
+        [Route("units")]
+        public async Task<IActionResult> GetUnits()
+        {
+            return Ok(await _unitServices.GetUnitsAsync());
+        }
+
+        [HttpGet]
+        [Route("units/{id}")]
```
Comment (Alex-knu) [inline MilitaryManager.Units.API/Controllers/UnitController.cs]: Rewrite [Route("{id}")]

File: MilitaryManager.Units.API/Controllers/UnitController.cs [hunk 3]
```diff
@@ -32,6 +32,20 @@ public async Task<IActionResult> GetTreeNode([FromRoute] int id)
             return Ok(await _unitServices.GetUnitsTreeAsync(id));
         }
 
+        [HttpGet]
+        [Route("units")]
```
Comment (Alex-knu) [inline MilitaryManager.Units.API/Controllers/UnitController.cs]: remove rout

File: MilitaryManager.Core/Services/UnitService.cs (hunk 1 above)
Comment (vladtymo) [reply MilitaryManager.Core/Services/UnitService.cs]: You can get just a unit without including related props.

File: MilitaryManager.Core/DTO/Attributes/AttributeDTO.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+using System.Text;
+
+namespace MilitaryManager.Core.DTO.Attributes
+{
+    public class AttributeDTO
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+
+        public List<ProfileDTO> Profiles { get; set; }
```
Comment (Alex-knu) [inline MilitaryManager.Core/DTO/Attributes/AttributeDTO.cs]: remove

File: MilitaryManager.Core/DTO/Attributes/AttributeDTO.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+        public List<EntityToAttributeDTO> EntityToAttributes { get; set; }
```
Comment (Alex-knu) [inline MilitaryManager.Core/DTO/Attributes/AttributeDTO.cs]: remove

File: MilitaryManager.Core/DTO/Entities/EntityDTO.cs [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+using System.Text;
+
+namespace MilitaryManager.Core.DTO.Entities
+{
+    public class EntityDTO
+    {
+        public int Id { get; set; }
+        public string RegNum { get; set; }
+
+        public List<EntityToAttributeDTO> EntityToAttributes { get; set; }
```
Comment (Alex-knu) [inline MilitaryManager.Core/DTO/Entities/EntityDTO.cs]: remove

File: MilitaryManager.Core/DTO/Entities/EntityDTO.cs [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+        public UnitToEquipmentDTO UnitToEquipment { get; set; }
```
Comment (Alex-knu) [inline MilitaryManager.Core/DTO/Entities/EntityDTO.cs]: remove

File: MilitaryManager.Core/DTO/Positions/PositionDTO.cs [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+using System.Text;
+
+namespace MilitaryManager.Core.DTO.Positions
+{
+    public class PositionDTO
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+
+        public List<UnitDTO> Units { get; set; }
```
Comment (Alex-knu) [inline MilitaryManager.Core/DTO/Positions/PositionDTO.cs]: remove

File: MilitaryManager.Core/DTO/Profiles/ProfileDTO.cs [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+{
+    public class ProfileDTO
+    {
+        public int Id { get; set; }
+        public int AttributeId { get; set; }
+        public int UnitId { get; set; }
+        public string Value { get; set; }
+
+        public AttributeDTO Attribute { get; set; }
+        public UnitDTO Unit { get; set; }
```
Comment (Alex-knu) [inline MilitaryManager.Core/DTO/Profiles/ProfileDTO.cs]: remove

File: MilitaryManager.Core/DTO/Ranks/RankDTO.cs [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+using System.Text;
+
+namespace MilitaryManager.Core.DTO.Ranks
+{
+    public class RankDTO
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+
+        public List<UnitDTO> Units { get; set; }
```
Comment (Alex-knu) [inline MilitaryManager.Core/DTO/Ranks/RankDTO.cs]: remove

**Review by Alex-knu — CHANGES_REQUESTED** [review]

**Review by vladtymo — APPROVED** [review]

File: MilitaryManager.Core/Services/UnitService.cs [hunk 11]
```diff
@@ -14,7 +17,8 @@ public class UnitService : IUnitService
         protected readonly IRepository<Unit, int> _unitRepository;
         protected readonly IMapper _mapper;
 
-        public UnitService(IRepository<Unit, int> unitRepository, IMapper mapper)
+        public UnitService(IRepository<Unit, int> unitRepository,
```
Comment (Alex-knu) [inline MilitaryManager.Core/Services/UnitService.cs]: ?

**Review by Alex-knu — CHANGES_REQUESTED** [review]

File: MilitaryManager.Core/Services/UnitService.cs (hunk 11 above)
Comment (mershalova) [reply MilitaryManager.Core/Services/UnitService.cs]: reverted this refactoring

**Review by Alex-knu — APPROVED** [review]

**Pull request merged**